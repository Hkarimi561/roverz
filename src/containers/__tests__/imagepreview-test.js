/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ImagePreview from '@containers/image/ImagePreview';
// import { Actions } from 'react-native-router-flux';


it('ImagePreview renders correctly', () => {
  const tree = renderer.create(
    <ImagePreview />,
    ).toJSON();
  expect(tree).toMatchSnapshot();
});


// it('Actions renders correctly', () => {
//   const cameraData = 'viswa';
//   const cameraMessage = 'message';
//   const expectedAction = {
//       // type: 'ADD_TODO',
//     cameraData: 'viswa',
//     cameraMessage: 'message',
//   };
//   expect(Actions.refresh(cameraData, cameraMessage)).toEqual(expectedAction);
// });

// it('Actions renders correctly', () => {
//  const attach = { cameraData: 'data', cameraMessage: 'message', video: true };
//   const expectedAction = {
//         // type: 'ADD_TODO',
//     cameraData: 'viswa',
//     cameraMessage: 'message',
//     video: 'true',
//   };
//  expect(Actions.refresh()).toEqual(attach);
// });

// it(' pop Actions renders correctly', () => {
//   expect(Actions.pop()).toEqual();
// });

