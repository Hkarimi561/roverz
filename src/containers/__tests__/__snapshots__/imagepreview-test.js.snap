// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePreview renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  />
  <View
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "left": 0,
            "padding": 16,
            "position": "absolute",
            "right": 0,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.4)",
            "bottom": 0,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Image caption.."
        style={
          Array [
            Object {
              "backgroundColor": "#F5F5F5",
              "borderRadius": 3,
              "flex": 1,
              "fontFamily": "OpenSans-Regular",
              "fontSize": 16,
              "height": 40,
              "marginRight": 8,
              "padding": 5,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 40,
            "opacity": 1,
            "width": 40,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          onPress={undefined}
          style={
            Array [
              false,
              false,
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          underlayColor="white"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#7377C9",
                  "fontSize": 34,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
      width={20}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 20,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`ImagePreview renders correctly 2`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  />
  <View
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "left": 0,
            "padding": 16,
            "position": "absolute",
            "right": 0,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.4)",
            "bottom": 0,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Image caption.."
        style={
          Array [
            Object {
              "backgroundColor": "#F5F5F5",
              "borderRadius": 3,
              "flex": 1,
              "fontFamily": "OpenSans-Regular",
              "fontSize": 16,
              "height": 40,
              "marginRight": 8,
              "padding": 5,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 40,
            "opacity": 1,
            "width": 40,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          onPress={undefined}
          style={
            Array [
              false,
              false,
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          underlayColor="white"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#7377C9",
                  "fontSize": 34,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
      width={20}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 20,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
